src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/ArrayMapNode.cs

```diff
@@ -57,12 +57,15 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
 
                 var arrayType = (ArrayType)type;
 
+                // This optimization is not compatible with canInlineTypeCheck on JIT/EE interface returning
```

Comment:
cc @MichalStrehovsky I have done this to make the test pass. The alternative was to change canInlineTypeCheck, but that would lead to suboptimal code in number of places.

Reply:
Makes sense. Thank you!